File: Data Structures & Algorithms/move-zeroes/submission-0.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        
        int j=-1;
        int i=0;
        while( i < nums.size() ) {
            if( nums[i] == 0 ) {
                j=i;
                break;
            }
            i++;
        }

        if( j == -1 ) {
            return;
        }
        for( int i=j; i<nums.size(); i++ ) {
            if( nums[i] != 0 ) {
                swap( nums[i] , nums[j]);
                j++;
            }
        }
    }
};

File: Data Structures & Algorithms/sort-array-by-parity/submission-0.java
class Solution {
    public int[] sortArrayByParity(int[] nums) {
        
        int[] arr = new int[nums.length];

        int j=0;
        for( int i=0; i<nums.length; i++ ) {
            if( nums[i] % 2 == 0 ) {
                arr[j++] = nums[i];
            }
        }
        for( int i=0; i<nums.length; i++ ) {
            if( nums[i] % 2 == 1 ) {
                arr[j++] = nums[i];
            }
        }

        return arr;
    }
}